ev3dev.py

```diff
@@ -2262,11 +2263,24 @@ def clear(self):
         """
         self._draw.rectangle(((0,0), self.shape), fill="white")
 
+    def _color565(self, r, g, b):
+	"""Convert red, green, blue components to a 16-bit 565 RGB value. Components
+	should be values 0 to 255.
+	"""
+	return (((r & 0xF8) << 8) | ((g & 0xFC) << 3) | (b >> 3))
+
+    def _img_to_rgb565_bytes(self):
+        pixels = [self._color565(r,g,b) for (r,g,b) in self._img.getdata()]
+        return struct.pack('H' * len(pixels), *pixels)
```

Comment:
Ok, it seems python array is just an efficient way to *store* data arrays; it does not provide any array manipulation functions. However, I measured performance of @dlech's list-based solution and PiStorms numpy-based one. numpy is about 64 times faster (on my desktop computer): http://nbviewer.ipython.org/gist/ddemidov/84053a30e443645fcbeb